sdk/azidentity/chained_token_credential.go

```diff
@@ -56,45 +56,39 @@ func NewChainedTokenCredential(sources []azcore.TokenCredential, options *Chaine
 // GetToken calls GetToken on the chained credentials in turn, stopping when one returns a token. This method is called automatically by Azure SDK clients.
 // ctx: Context controlling the request lifetime.
 // opts: Options for the token request, in particular the desired scope of the access token.
-func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (token *azcore.AccessToken, err error) {
+func (c *ChainedTokenCredential) GetToken(ctx context.Context, opts policy.TokenRequestOptions) (*azcore.AccessToken, error) {
 	if c.successfulCredential != nil && !c.retrySources {
 		return c.successfulCredential.GetToken(ctx, opts)
 	}
-	var errList []credentialUnavailableError
+
+	var errs []error
 	for _, cred := range c.sources {
-		token, err = cred.GetToken(ctx, opts)
-		var credErr credentialUnavailableError
-		if errors.As(err, &credErr) {
-			errList = append(errList, credErr)
-		} else if err != nil {
-			var authFailed AuthenticationFailedError
-			if errors.As(err, &authFailed) {
-				err = fmt.Errorf("%s: %s\n\t%s", c.name, createChainedErrorMessage(errList), err)
-				authErr := newAuthenticationFailedError(err, authFailed.RawResponse)
-				return nil, authErr
-			}
-			return nil, err
-		} else {
+		token, err := cred.GetToken(ctx, opts)
+		if err == nil {
 			log.Writef(EventAuthentication, "Azure Identity => %s authenticated with %s", c.name, extractCredentialName(cred))
 			c.successfulCredential = cred
 			return token, nil
 		}
+		errs = append(errs, err)
+		if _, ok := err.(credentialUnavailableError); !ok {
```

Comment:
Shouldn't this be checking for `authenticationFailedError`?

Reply:
The only expected, non-fatal error type is `credentialUnavailableError`. If we have any other type of error, I want to return `AuthenticationFailedError`.

Reply:
Is there actually an HTTP response in this case though?  I thought that we get this error if there's not enough info to even attempt to authenticate.

Reply:
Ooops I just realized the condition is `!ok`, makes sense now.